tests/dialects/test_tsql.py

```diff
@@ -13,6 +13,30 @@ def test_tsql(self):
         self.validate_identity('SELECT "x"."y" FROM foo')
         self.validate_identity("SELECT * FROM #foo")
         self.validate_identity("SELECT * FROM ##foo")
+        self.validate_identity("DECLARE @TestVar nvarchar(MAX)")
+        self.validate_identity("""
+        CREATE FUNCTION [TRANSF].[FS_Tester]
+(
+	@Offer nvarchar(2000)
+)
+RETURNS nvarchar(2000)
+AS
+BEGIN
+	RETURN '1';
+END
+
+        """)
+        self.validate_identity("""
+        CREATE procedure [TRANSF].[SP_DO_STUFF]
+
+   @Loadid INT 
+  --,@YearMonth int=0
+  ,@NumberOfRows INT = NULL OUT
+AS 
+BEGIN 
+    PRINT 'hi'
+END
+        """)
```

Comment:
You should use `validate_all` or set the pretty parameter here; SQLGlot doesn't format by default, so your SQL will be transformed in a single line. Look at some other tests for reference and let me know if anything's unclear.